core/src/main/java/org/pentaho/platform/util/oauth/PentahoOAuthUtility.java

```diff
@@ -0,0 +1,214 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.platform.util.oauth;
+
+import org.apache.commons.lang.StringUtils;
+import org.pentaho.platform.api.engine.IConfiguration;
+import org.pentaho.platform.api.engine.ISystemConfig;
+import org.pentaho.platform.engine.core.system.PentahoSystem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.client.RestTemplate;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+public class PentahoOAuthUtility {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger( PentahoOAuthUtility.class );
+
+  public static final String OAUTH = "oauth";
+
+  public static final String DUALAUTH = "dualauth";
+
+  public static final String REGISTRATION_ID = "registrationId";
+
+  public static final String USER_ID = "userId";
+
+  RestTemplate restTemplate = new RestTemplate();
+
+  PentahoOAuthProperties pentahoOAuthProperties = new PentahoOAuthProperties();
+
+  private static ISystemConfig systemConfig = PentahoSystem.get( ISystemConfig.class );
+
+  private static boolean isProviderOAuth = false;
+
+  private static boolean isProviderDualAuth = false;
+
+  private static PentahoOAuthUtility instance;
+
+  // Private constructor to prevent instantiation from outside
+  private PentahoOAuthUtility() {
+  }
+
+  // Public method to provide access to the single instance
+  public static PentahoOAuthUtility getInstance() {
```

Comment:
Not a blocker, 
why do you need to make this class a singleton via` getInstance()`. when we have spring in PUC, which manage the creation of classes via scope - https://www.baeldung.com/spring-bean-scopes#singleton ?

Reply:
addressed.. not with spring scopes of bean needed for this class